### PR TITLE
Resolve #137 Relaxed HomeBrew requirement

### LINKED ISSUE #137: Relaxed HomeBrew requirement
When I don't have HomeBrew, but dependencies are met, I'm still having problems running it due to a (non-critical?) dependency check:

```
2015-11-11 17:03:25 [ERROR] This script requires HomeBrew, but it's not installed. Aborting.
```

I'd propose relaxing the HomeBrew requirement and applying the check only in the install call.

### PR DESCRIPTION
**Pull request by ain**
Check against HomeBrew only if missing dependencies need to be installed

### CODE REVIEW

**Comment by ain** [comment]
@brikis98 I'd appreciate if you could give it a run on HomeBrew as well, to double-check.

**Comment by brikis98** [comment]
Apologies, I'm a bit overloaded this week. The code looks good, but I don't have time to try the PR locally to see if it works well with HomeBrew. I'll try to get to it in a couple days.

File: src/docker-osx-dev [hunk 1]
```diff
@@ -627,16 +622,14 @@ function brew_install {
...
-  fi
+  if ! type "$command_name" > /dev/null 2>&1 ; then
+    check_homebrew $readable_name
+
+    local brew_command="brew"
+    if [[ "$use_cask" = true ]]; then
+      brew_command="brew cask"
+    fi
 
-  if eval "$brew_command list $package_name" > /dev/null 2>&1 ; then
```
Comment (brikis98) [inline src/docker-osx-dev:635]: Just ran this locally, and I think you lost this check. As a result, it tried to install Brew Cask, which I already had installed, and exited with an error.

Comment (ain) [inline src/docker-osx-dev:635]: Thanks, will fix.

**Comment by ain** [comment]
@brikis98 adjusted, rebased.

**Comment by brikis98** [comment]
Just tried it locally, and there are two issues:
1. The script runs `brew update` before all the calls to `brew_install`. Obviously, this will fail for anyone without HomeBrew.
2. I wanted to check the script was idempotent and ran it on my computer, which already has all dependencies installed. It failed with this error:

```
2015-11-22 01:00:44 [WARN] Cask is already installed by HomeBrew, skipping
2015-11-22 01:00:44 [ERROR] Failed to install dependency Cask. Check the log output above for reasons.
```

Something about the "is already installed" check is failing...

**Comment by ain** [comment]
@brikis98 thanks for feedback. I've wrapped things around a bit, I think the latest commit also brings better readability and encapsulation.

There's a connected topic of Cask being a breaking dependency (see #146) that I'd question as well, but deal with in a separate PR once this one gets merged.

File: src/docker-osx-dev [hunk 2]
```diff
@@ -600,10 +600,19 @@ function check_prerequisites {
...
+# Update Brew formulae and HomeBrew
+#
+# Non-breaking. Falls through with warning if fails.
+#
+function brew_update {
+  log_info "Updating HomeBrew"
   if ! type brew > /dev/null 2>&1 ; then
-    log_error "This script requires HomeBrew, but it's not installed. Aborting."
-    exit 1
+    log_warn "HomeBrew not found, attempting to continue without"
```
Comment (brikis98) [inline src/docker-osx-dev:613]: Just for clarity, may want to have the comment say "HomeBrew not found, will not attempt to update it".

**Comment by brikis98** [comment]
Just ran it locally and in a (not very comprehensive) test, it seems to work fine. So other than the minor comment, this is good to go.

**Comment by brikis98** [comment]
In fact, I just merged it in. You can update that warning text in a separate commit, w/o a PR, as it's a very minor thing.

**Comment by ain** [comment]
Thanks Yevgeniy, I'll have another PR on your suggestion, makes sense.

**Pull request merged**